### PR TITLE
Refactored MessageType enum

### PR DESCRIPTION
**Pull request by Kardbord**
Eliminated the `MessageType.VALUE` attribute; replaced with `toShort` method based on the enumerated type's ordinal.

Implemented `MessageType::getTypeFromShort` method to map from shorts to MessageType.

These changes are intended to make the `MessageType` enumeration more easily extendable should the need arise. As long as all processes use the same `Message` class, and as long as MessageType short values are **never** hard-coded, these changes should be beneficial. MessageType short values should always be retrieved and/or set by using `MessageType.<MessageType>.toShort()`

### CODE REVIEW

File: messages/src/main/java/Messages/Message.java [hunk 1]
```diff
@@ -10,29 +10,30 @@
...
-        CANCEL_ORDER             ((short)  7),
-        FWD_CANCEL               ((short)  8),
-        CANCEL_CONF              ((short)  9),
-        FWD_CANCEL_CONF          ((short) 10),
-        REGISTER_MATCHING_ENGINE ((short) 11),
-        ACK                      ((short) 12),
-        TOP_OF_BOOK_NOTIFICATION ((short) 13),
-        TOP_OF_BOOK_REQUEST      ((short) 14),
-        TOP_OF_BOOK_RESPONSE     ((short) 15),
+        UNKNOWN                  ,
```
Comment (CarsonBradshaw) [inline messages/src/main/java/Messages/Message.java]: Style Nit: Commas should be left aligned against Enums. It's very likely that our linter picks it up (and automatically fixes it) tomorrow when we set it up but good to get in the habit.

Comment (Kardbord) [reply messages/src/main/java/Messages/Message.java]: I'll give it a fix real quick, as I want to make one other change anyway. It occurred to me that I ought to do some bounds checking before trying to map a short to a value.

**Comment by Kardbord** [comment]
As mentioned in another comment, it occurred to me that some bounds checking should occur in the `getTypeFromShort` method. Any strong opinions from anybody as to what should be done if the provided ordinal is out of bounds? The way I see it, there are three options:
1. Do no bounds checking - let the ArrayIndexOutOfBounds exception filter up
2. Throw our own exception if the ordinal is out of range
3. Return `null` if the ordinal is out of range

**Comment by CarsonBradshaw** [comment]
@Kardbord I vote option 1. Option 2 would be saying essentially what `ArrayIndexOutOfBounds` already does. Option 3 could lead to more errors down the line. As a general rule, it is best to avoid returning null when you can.

File: messages/src/main/java/Messages/Message.java (hunk 1 above)
Comment (Kardbord) [reply messages/src/main/java/Messages/Message.java]: Made those style fixes.

**Pull request merged**